src/devices/sound/c140.cpp

```diff
@@ -158,12 +161,57 @@ void c140_device::device_start()
 	}
 }
 
+void c219_device::device_start()
+{
+	c140_device::device_start();
+	memset(m_c219bank, 0, sizeof(m_c219bank));
 
-//-------------------------------------------------
-//  sound_stream_update - handle a stream update
-//-------------------------------------------------
+	save_item(NAME(m_c219bank));
+}
 
-void c140_device::sound_stream_update(sound_stream &stream, stream_sample_t **inputs, stream_sample_t **outputs, int samples)
+void c140_device::device_reset()
+{
+	memset(m_REG,0,sizeof(m_REG));
+
+	for(int i = 0; i < MAX_VOICE; i++)
+	{
+		init_voice(&m_voi[i]);
+	}
+}
+
+void c219_device::device_reset()
+{
+	c140_device::device_reset();
+	memset(m_c219bank, 0, sizeof(m_c219bank));
+}
+
+void c140_device::device_clock_changed()
+{
+	int old_rate = m_sample_rate;
+	m_sample_rate = m_baserate = clock();
+	if (old_rate < m_sample_rate)
+	{
+		memset(&m_mixer_buffer_left[0], 0, sizeof(m_mixer_buffer_left));
+		memset(&m_mixer_buffer_right[0], 0, sizeof(m_mixer_buffer_right));
```

Comment:
Here's a great example of why `std::fill` is a better option - `sizeof` returns the size of the `std::vector` object and bears no relation to the size of its content.